server/src/main/java/org/elasticsearch/index/mapper/MetadataFieldMapper.java

```diff
@@ -144,6 +150,22 @@ public final void parseMetadataField(String name, MappingParserContext parserCon
                 final Object propNode = entry.getValue();
                 Parameter<?> parameter = paramsMap.get(propName);
                 if (parameter == null) {
+                    IndexVersion indexVersionCreated = parserContext.indexVersionCreated();
+                    if (indexVersionCreated.before(IndexVersions.UPGRADE_TO_LUCENE_10_0_0)
```

Comment:
I'm not entirely sure if UPGRADE_TO_LUCENE_10_0_0 should be the point in time we should start to throw errors. We first removed support for these parameters with #116944 (https://github.com/elastic/elasticsearch/commit/8a7491cdb89c75e61120ebca7f7c16d1477e1f5c) at which point we already have more recent IndexVersions (see https://github.com/elastic/elasticsearch/blob/8a7491cdb89c75e61120ebca7f7c16d1477e1f5c/server/src/main/java/org/elasticsearch/index/IndexVersions.java#L135). To me that means indices with e.g. IndexVersion TIME_BASED_K_ORDERED_DOC_ID could still use those parameters? Please keep me honest here.

Reply:
good point. Tricky question. I don't expect it to matter a lot because these unsupported params should not be common at all. The removal affects only serverless users indeed, but I'm afraid it can't be associated to a specific index version. If we meant to do so we'd have needed to add an index versions associated with the removal?

I am fine with using what you are using, or in alternative adding a new index version now.